File: week5/Medium.java
public class Medium {

    private static boolean isPrime(int num) {
        if (num < 2) return false;

        for (int i = 2; i < num; i++) {
            if (num % i == 0) return false;
        }

        return true;
    }

    private static int largestPrimeUnder(int num) {
        for (int i = num; i > 1; i--) {
            if (isPrime(i)) return i;
        }

        // No primes under this number?
        return -1;
    }

    public static void main(String[] args) {

        System.out.println(largestPrimeUnder(1_000_000));

    }

}

File: week5/medium.py
def is_prime(num: int) -> bool:
    if num < 2:
        return False
    
    for i in range(2, num):
        if num % i == 0:
            return False
    
    return True


def largest_prime_under(num: int) -> int:
    for i in range(num, 1, -1):
        if is_prime(i):
            return i
    
    # No primes under this number?
    return -1

print(largest_prime_under(1_000_000))

